src/Ouroboros-Mini_Protocols-Chain_Sync.thy

```diff
@@ -0,0 +1,178 @@
+section \<open>Chain Synchronization Mini-Protocol\<close>
+
+text \<open>
+  The theory in this section implements the chain synchronization mini-protocol.
+\<close>
+
+theory "Ouroboros-Mini_Protocols-Chain_Sync"
+  imports
+    "Ouroboros-Mini_Protocols"
+    "HOL-Library.BNF_Corec"
+begin
+
+locale chain_sync =
+  fixes initial_items\<^sub>c :: "'i list"
+  fixes initial_items\<^sub>s :: "'i list"
+  fixes candidate_points :: "'i list \<Rightarrow> 'p list"
+  fixes best_intersection_point :: "'i list \<Rightarrow> 'p list \<Rightarrow> 'p option"
+  fixes point_of :: "'i \<Rightarrow> 'p"
+
```

Comment:
I think that it would be good to insert a bit of documentation here that at least says that `'i` stands for “item” and what we mean by “item”. Also telling the meaning of `'p` would be good.

Reply:
Fixed in bd02e925c353588f73d5415cb8e3809bb9443114, please check.

Reply:
I have some idea regarding how to slightly polish this, but this can wait until my final, proof-reading round. :slightly_smiling_face: